File: 은지/LV2/거리두기 확인하기.py
def solution(places):
    answer = []
    for p in places:
        person = []
        for i in range(5):
            for j in range(5):
                if p[i][j] == "P":
                    person.append((i, j))
        
        cnt = 1
        for i in range(len(person)-1):
            for j in range(i+1, len(person)):
                r1 = person[i][0]
                r2 = person[j][0]
                c1 = person[i][1]
                c2 = person[j][1]
                if r1 == r2:
                    if abs(c1-c2) == 1:
                        cnt = 0
                        break
                    elif abs(c1-c2) == 2:
                        if p[r1][(c1+c2)//2] != "X":
                            cnt = 0
                            break
                elif c1 == c2:
                    if abs(r1-r2) == 1:
                        cnt = 0
                        break
                    elif abs(r1-r2) == 2:
                        if p[(r1+r2)//2][c1] != "X":
                            cnt = 0
                            break
                # 대각선 확인
                else:
                    if abs(r1-r2) == 1 and abs(c1-c2) == 1:
                        if p[r1][c2] != "X" or p[r2][c1] != "X":
                            cnt = 0
                            break

        if cnt:
            answer.append(1)
        else:
            answer.append(0)
    return answer

File: 은지/LV2/거리두기 확인하기.js
function solution(places) {
  var answer = [];
  for (let p of places) {
    var person = []
    for (let i=0; i<5; i++) {
      for (let j=0; j<5; j++) {
        if (p[i][j] === "P") {
          person.push([i, j])
        }
      }
    } 
    var cnt = 1
    while (person.length) {
      const v = person.shift()

      function chk(r, c) {
        // 1칸 확인
        if (r+1 < 5 && p[r+1][c] === "P") {
          return 0
        }
        if (c+1 < 5 && p[r][c+1] === "P") {
          return 0
        }
        // 2칸 확인
        if (r+2 < 5 && p[r+2][c] === "P" && p[r+1][c] !== "X") {
          return 0
        }
        if (c+2 < 5 && p[r][c+2] === "P" && p[r][c+1] !== "X") {
          return 0
        }
        // 대각선 확인
        if (0 <= r-1 && c+1 < 5 && p[r-1][c+1] === "P") {
          if (p[r-1][c] !== "X" || p[r][c+1] !== "X") {
            return 0
          }
        }
        if (r+1 < 5 && c+1 < 5 && p[r+1][c+1] === "P") {
          if (p[r+1][c] !== "X" || p[r][c+1] !== "X") {
            return 0
          }
        }
      }

      var chk_v = chk(v[0], v[1])
      
      if (chk_v == 0) {
        cnt = 0
        break
      }
    }
    if (cnt) {
      answer.push(1)
    } else {
      answer.push(0)
    }
  }
  return answer;
}